### PR TITLE
feat(chain manager): add as many transactions as fit in block

### PR DESCRIPTION
**Pull request by ghost**
Store as many transactions as they can fit when building a block, using the weight parameter.

### CODE REVIEW

File: core/src/actors/chain_manager/mod.rs [hunk 1]
```diff
@@ -226,15 +228,28 @@ impl ChainManager {
...
-                // TODO: t.fee()
-                transaction_fees += 1;
-                *t
-            })
-            .collect();
+        let max_block_weight = self.max_block_weight;
+        let mut block_weight = 0;
+        let mut transactions = Vec::new();
+
+        for transaction in self.transactions_pool.iter() {
```
Comment (Tommytrg) [inline core/src/actors/chain_manager/mod.rs]: Not functional enough

Comment (anler) [reply core/src/actors/chain_manager/mod.rs]: Nah, the thing is that I added a small optimization: if the weight of the included transactions is equal to the maximum weight allowed for the block, then there's no need to keep processing more transactions so a `break` is used to stop the loop.

File: config/src/defaults.rs [hunk 2]
```diff
@@ -91,6 +91,11 @@ pub trait Defaults {
 
     /// Default JSON-RPC server addr
     fn jsonrpc_server_address(&self) -> SocketAddr;
+
+    fn consensus_constants_max_block_weight(&self) -> u32 {
+        // TODO: Replace with real max_block_weight value used in mainnet
```
Comment (Tommytrg) [inline config/src/defaults.rs]: Replace TODO by FIXME(#issue_number) to be able to tack this.

File: core/src/actors/chain_manager/mod.rs [hunk 3]
```diff
@@ -225,17 +227,29 @@ impl ChainManager {
...
+
+        let max_block_weight = self.max_block_weight;
+        let mut block_weight = 0;
+        let mut transactions = Vec::new();
+
+        for transaction in self.transactions_pool.iter() {
+            // Currently, 1 weight unit is equivalent to 1 byte
+            let transaction_weight = transaction.size();
+            // TODO: Calculate transaction fee
+            let transaction_fee = 1;
```
Comment (aesedepece) [inline core/src/actors/chain_manager/mod.rs]: Let's take this chance to create `transaction.fee()` returning a static `1`. Then incorporating the actual fee calculation logic will be plug-and-play.

File: data_structures/src/chain.rs [hunk 4]
```diff
@@ -94,6 +105,13 @@ pub struct Block {
     pub txns: Vec<Transaction>,
 }
 
+/// Any reference to a Hashable type is also Hashable
+impl<'a, T: Hashable> Hashable for &'a T {
```
Comment (aesedepece) [inline data_structures/src/chain.rs]: This is wonderful :heart_eyes:

**Pull request merged**